### PR TITLE
Login page

### PR DESCRIPTION
**Pull request by foolhardy21**
## Login Page

- This PR has login page components and APIs.

- Files to review:
   - [ ] src/pages/Login.js
   - [ ] src/components/Login/LoginForm.js
   - [ ] src/contexts/login.context.js
   - [ ] src/contexts/auth.context.js
   - [ ] src/reducers/login.reducer.js

v-1

- Use this link for preview - https://deploy-preview-2--pensive-hodgkin-f2cf81.netlify.app/login

Thanks.

### CODE REVIEW

File: src/stylesheet/media-queries.css [hunk 1]
```diff
@@ -0,0 +1,40 @@
+@media only screen and (max-width: 1024px) {
+    /* .grid-itm2 {
+        grid-row: 1/2;
+    }
+    
+    .grid-itm1 {
+        grid-row: 2/3;
+    } */
+}
```
Comment (vrinda-mahajan) [inline src/stylesheet/media-queries.css]: remove the commented code

**Review by vrinda-mahajan — APPROVED** [review]
In the signup form, it should be written confirm password instead of password in the last field.
Screenshot (98)

Suggestion - you can reduce the contrast of your text in dark mode by selecting a different shade of black and white color.
Other than this code looks good and validation functionality is also working properly.

File: src/components/Login/LoginForm.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+import { getSolidBtnBgColor, getSolidBtnTextColor } from "../../utils"
+import { useLogin, useTheme, useAuth } from '../../contexts'
+
+const LoginForm = () => {
+    const navigate = useNavigate()
+    const location = useLocation()
+    const { theme } = useTheme()
+    const { loginFormState, isFormInvalid, loginFormDispatch, showLoginAlert } = useLogin()
+    const { loginUser } = useAuth()
+    const from = location.state?.from?.pathname || '/videos'
```
Comment (nayyyhaa) [inline src/components/Login/LoginForm.js:12]: Appreciation: good use of optional chaining here

File: src/components/Reusable/Button.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+const Button = ({ onClick, id, classes, children }) => {
+
+    return (
+        <button id={id} onClick={onClick} className={classes}>
```
Comment (nayyyhaa) [inline src/components/Reusable/Button.js]: good that you are passing custom `onClick` function here in component for reusability.

File: src/components/Login/LoginHeader.js [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+                        history
+                    </Icon>
+                </Link>
+
+
+                <Link to='/signup'>
+                    <Button classes={`btn-txt txt-lcase ${getTextColor(theme)} pd-xs txt-md`}>
+                        signup
+                    </Button>
+                </Link>
```
Comment (nayyyhaa) [inline src/components/Login/LoginHeader.js:50]: this part (..cont)

File: src/components/Signup/SignupHeader.js [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+                <Link to='/history'>
+                    <Icon classes={`${getIconColor(theme)} mg-right-lg`}>
+                        history
+                    </Icon>
+                </Link>
+
+                <Link to='/login'>
+                    <Button classes={`btn-txt txt-lcase ${getTextColor(theme)} bg-primary pd-xs txt-md`}>
+                        login
+                    </Button>
```
Comment (nayyyhaa) [inline src/components/Signup/SignupHeader.js]: (..cont) + this part.
both login + signup can reuse the same header component. You can conditionally render login/signup button based on the path...

**Review by nayyyhaa — APPROVED** [review]
I've added a suggestion for header component reusability. Rest the code looks good.

File: src/components/Login/LoginForm.js (hunk 2 above)
Comment (foolhardy21) [reply src/components/Login/LoginForm.js:12]: thankyou

File: src/components/Signup/SignupHeader.js (hunk 5 above)
Comment (foolhardy21) [reply src/components/Signup/SignupHeader.js]: thanks for this feedback. I'll surely include it.

File: src/stylesheet/media-queries.css (hunk 1 above)
Comment (foolhardy21) [reply src/stylesheet/media-queries.css]: made the changes. thanks.

**Pull request merged**